jslib/fetchDataStream.d.ts

```diff
@@ -0,0 +1,22 @@
+/**
+ * fetchDataStream fetches a data stream from dataURL where
+ * rows are denoted by the byte 0x1f, columns by 0x1e, and
+ * the stream is terminated by 0x04.
+ *
+ * @param {string} dataURL   - the URL of the stream to fetch
+ * @param {function} rowcb   - A callback to call for each row
+ * @param {function} chunkcb - A callback to call for each chunk
+ *                             read from the stream.
+ * @param {function} endstreamcb - A callback to call when the final
+ *                             byte is read from the stream.
+ * @param {string} method    - the HTTP method to use for the request
+ */
+async function fetchDataStream(
+    dataURL: string,
+    rowcb: (row: string[]) => void,
+    chunkcb: () => void,
+    endstreamcb: () => void,
+    method: string
```

Comment:
Might be good to switch for an enum: 'GET' | 'PUT' | 'POST' | 'PATCH' | 'DELETE'

Reply:
I don't see this param in the function def: https://github.com/aces/Loris/blob/main/jslib/fetchDataStream.js#L45

Reply:
Hm.. you're right. I wrote the .d.ts file for this: https://github.com/aces/Loris/pull/8907/files#diff-7409e296bbfa103b243c29d76808eae3ed3a59d2b4ec00498e7607b05858a457L56-L60

Do you want me to add the change to the function to this PR too?

Reply:
I see. That's not necessary, I assume you will do the merges ion the appropriate order :)

Reply:
I would prefer to do it here to keep them in sync, I'll port it over